# Sprint 4 Project EDA

The purpose of this notebook is to perform EDA on some car advertisement data to see what sort of graphs would be interesting to make for a web application.

## Set Up Environment and Data

### Import and Read

In [82]:
# import pkgs
import pandas as pd
import plotly.express as px
import altair as alt
import streamlit as st

In [83]:
# read data
car_df = pd.read_csv('~/TripleTen/Sprint_4/Sprint_4_Project/vehicles_us.csv')
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    47906 non-null  float64
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     46265 non-null  float64
 5   fuel          51525 non-null  object 
 6   odometer      43633 non-null  float64
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   42258 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(4), int64(2), object(7)
memory usage: 5.1+ MB


In [84]:
car_df.sample(50)

price  model_year                     model  condition  cylinders  \
48803   4995         NaN            toyota 4runner       good        8.0   
15651   3000      2005.0         gmc sierra 2500hd       fair        8.0   
31737   5490      2008.0                ford f-150       good        6.0   
24978   9750      2012.0  chevrolet silverado 1500   like new        8.0   
32106   6500      1994.0             jeep wrangler  excellent        4.0   
4944    9399      2015.0             ford focus se   like new        4.0   
33229   9999      2003.0         cadillac escalade       good        8.0   
11985  33995      2013.0   jeep wrangler unlimited  excellent        6.0   
16547  15500      2017.0               ford escape       good        4.0   
33165   6495      2012.0              toyota prius  excellent        4.0   
13772   6995      2005.0              dodge dakota  excellent        8.0   
6968   32988      2016.0           chevrolet tahoe   like new        8.0   
14781   5750      2014.0             ford focus se       good        4.0   
15439  16500      2013.0       chevrolet silverado       good        8.0   
46935  24825      2017.0              nissan rogue   like new        4.0   
16982   2300      1996.0               ford ranger       good        6.0   
36083   7750      2011.0           toyota camry le  excellent        4.0   
22552   5900      2001.0       chevrolet silverado  excellent        8.0   
26711   7990      2006.0            ford econoline       good        8.0   
7355   11500      2014.0               honda civic       good        4.0   
25986  13980      2018.0            hyundai sonata  excellent        NaN   
37809  24000      2005.0         gmc sierra 2500hd  excellent        8.0   
1490    4500      2012.0          volkswagen jetta  excellent        4.0   
35498  23950      2015.0                gmc sierra  excellent        8.0   
3103   21900      2013.0  ford mustang gt coupe 2d       good        NaN   
40417   9999      2004.0             jeep wrangler  excellent        6.0   
11290   6490      2012.0           toyota camry le  excellent        4.0   
10894   9500      2015.0            hyundai sonata  excellent        NaN   
49026  10995      2011.0  chevrolet silverado 1500       good        NaN   
20221  10995      2017.0             nissan altima  excellent        4.0   
43179  11000      2007.0       chevrolet silverado   like new        8.0   
9811    5995      2004.0           chevrolet tahoe       good        8.0   
13843  18990      2012.0                  ram 1500  excellent        8.0   
20288   4795         NaN           toyota camry le       fair        4.0   
14224   3995      2012.0            honda civic lx       good        4.0   
11522    800      2000.0                ford focus       fair        4.0   
30060   9450      2015.0           nissan frontier  excellent        6.0   
20545   8495      2012.0               honda civic       good        4.0   
7854    3200      2004.0        chevrolet suburban       good        8.0   
29894  11000      2008.0             toyota tacoma       good        6.0   
40964   6599      2007.0              honda accord  excellent        6.0   
31782   9900      2014.0               ford fusion  excellent        4.0   
42024   6000      2007.0                ford f-150  excellent        8.0   
36780   9900      2016.0                ford f-250       fair        8.0   
26465  16400      2011.0         cadillac escalade       good        8.0   
8289   12000      2003.0             jeep wrangler       good        6.0   
6187    6500      2012.0           chevrolet cruze  excellent        4.0   
26381  29900      2012.0             ford f-350 sd       good        8.0   
49063   8950      2013.0             nissan altima  excellent        NaN   
34005  32000      2017.0                ford f-150  excellent        NaN   

         fuel  odometer transmission         type paint_color  is_4wd  \
48803     gas  250000.0    automatic          SUV        

Some ideas: parse out make and model separately; parse out post year, month, day separately.

### Clean Up Duplicates and Nulls

In [85]:
# Check for Dups
car_df.duplicated().unique()

array([False])

In [86]:
# Clean up price column
car_df['price'].describe()

count     51525.000000
mean      12132.464920
std       10040.803015
min           1.000000
25%        5000.000000
50%        9000.000000
75%       16839.000000
max      375000.000000
Name: price, dtype: float64

In [87]:
car_df[car_df['price'] == 1]['price'].count()

798

It is unknown what a price of 1 means. It could be that some of these cars were sold at 1 dollar, but not likely 798 of them.

In [88]:
car_df.groupby('price')['cylinders'].value_counts().head(20)

price  cylinders
1      10.0         295
       6.0          172
       8.0          171
       4.0           88
       3.0            1
3      6.0            1
5      6.0            1
6      8.0            1
9      4.0            1
10     4.0            1
11     8.0            1
12     4.0            2
       6.0            1
15     8.0            3
20     8.0            1
24     10.0           1
25     6.0            1
       8.0            1
28     8.0            1
32     4.0            1
Name: count, dtype: int64

As suspected, the sale of cars at such low prices seems very uncommon. Articles on the subject say that online marketplaces have sellers with such listings as a search trick, to get their car to the top of the cheapest car search list, and then in the sale description confess this isn't the actual price of the car, it's a "listing fee" or something like that (the algorithm does catch some of these and removes them). Perusing the dates posted and days listed columns suggest that these cars are not all sold on the same day, these are just listings, so the 1 dollar trick is probably at play here. Take Note: a histogram of overall prices might be good to look at to see how many of these listings there are.

The model year is pretty straightforward, there are a few nulls. This most likely means the model year is unknown, but it won't do to replace these nulls with a string, it should be replaced with an integer that won't be confused with a known model year. They didn't drive cars in year 1.

In [89]:
car_df['model_year'] = car_df['model_year'].fillna(1).astype('int64')
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    51525 non-null  int64  
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     46265 non-null  float64
 5   fuel          51525 non-null  object 
 6   odometer      43633 non-null  float64
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   42258 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(3), int64(3), object(7)
memory usage: 5.1+ MB


In [90]:
car_df['cylinders'].value_counts()

cylinders
8.0     15844
6.0     15700
4.0     13864
10.0      549
5.0       272
3.0        34
12.0        2
Name: count, dtype: int64

Likewise, there are no cars with a 1-cylinder engine.

In [91]:
car_df['cylinders'] = car_df['cylinders'].fillna(1).astype('int64')
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    51525 non-null  int64  
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     51525 non-null  int64  
 5   fuel          51525 non-null  object 
 6   odometer      43633 non-null  float64
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   42258 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(2), int64(4), object(7)
memory usage: 5.1+ MB


In [92]:
car_df['odometer'].describe()

count     43633.000000
mean     115553.461738
std       65094.611341
min           0.000000
25%       70000.000000
50%      113000.000000
75%      155000.000000
max      990000.000000
Name: odometer, dtype: float64

Let's set the unlisted odometer readings to -1.

In [93]:
car_df['odometer'] = car_df['odometer'].fillna(-1).astype('int64')
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    51525 non-null  int64  
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     51525 non-null  int64  
 5   fuel          51525 non-null  object 
 6   odometer      51525 non-null  int64  
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   42258 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(1), int64(5), object(7)
memory usage: 5.1+ MB


Paint color is all strings, and should be so, thus the nulls should be read as unlisted.

In [94]:
car_df['paint_color'] = car_df['paint_color'].fillna('unlisted')
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    51525 non-null  int64  
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     51525 non-null  int64  
 5   fuel          51525 non-null  object 
 6   odometer      51525 non-null  int64  
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   51525 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(1), int64(5), object(7)
memory usage: 5.1+ MB


In [95]:
car_df['is_4wd'].value_counts()

is_4wd
1.0    25572
Name: count, dtype: int64

The nulls could mean either that the car is not 4wd, or it is unlisted. These should be converted to zero.

In [96]:
car_df['is_4wd'] = car_df['is_4wd'].fillna(0).astype('int64')
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   price         51525 non-null  int64 
 1   model_year    51525 non-null  int64 
 2   model         51525 non-null  object
 3   condition     51525 non-null  object
 4   cylinders     51525 non-null  int64 
 5   fuel          51525 non-null  object
 6   odometer      51525 non-null  int64 
 7   transmission  51525 non-null  object
 8   type          51525 non-null  object
 9   paint_color   51525 non-null  object
 10  is_4wd        51525 non-null  int64 
 11  date_posted   51525 non-null  object
 12  days_listed   51525 non-null  int64 
dtypes: int64(6), object(7)
memory usage: 5.1+ MB


### Sophisticate Data

In [97]:
#parse out make and model
car_df['make'] = car_df['model'].str.split().str[0]
car_df['make'].value_counts()

make
ford             12672
chevrolet        10611
toyota            5445
honda             3485
ram               3316
jeep              3281
nissan            3208
gmc               2378
subaru            1272
dodge             1255
hyundai           1173
volkswagen         869
chrysler           838
kia                585
cadillac           322
buick              271
bmw                267
acura              236
mercedes-benz       41
Name: count, dtype: int64

In [98]:
#Parse out Year, Month, Day
car_df['date_posted'] = pd.to_datetime(car_df['date_posted'])
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   price         51525 non-null  int64         
 1   model_year    51525 non-null  int64         
 2   model         51525 non-null  object        
 3   condition     51525 non-null  object        
 4   cylinders     51525 non-null  int64         
 5   fuel          51525 non-null  object        
 6   odometer      51525 non-null  int64         
 7   transmission  51525 non-null  object        
 8   type          51525 non-null  object        
 9   paint_color   51525 non-null  object        
 10  is_4wd        51525 non-null  int64         
 11  date_posted   51525 non-null  datetime64[ns]
 12  days_listed   51525 non-null  int64         
 13  make          51525 non-null  object        
dtypes: datetime64[ns](1), int64(6), object(7)
memory usage: 5.5+ MB


In [99]:
car_df['year_posted'] = car_df['date_posted'].dt.year
car_df['month_posted'] = car_df['date_posted'].dt.month
car_df['day_posted'] = car_df['date_posted'].dt.day
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   price         51525 non-null  int64         
 1   model_year    51525 non-null  int64         
 2   model         51525 non-null  object        
 3   condition     51525 non-null  object        
 4   cylinders     51525 non-null  int64         
 5   fuel          51525 non-null  object        
 6   odometer      51525 non-null  int64         
 7   transmission  51525 non-null  object        
 8   type          51525 non-null  object        
 9   paint_color   51525 non-null  object        
 10  is_4wd        51525 non-null  int64         
 11  date_posted   51525 non-null  datetime64[ns]
 12  days_listed   51525 non-null  int64         
 13  make          51525 non-null  object        
 14  year_posted   51525 non-null  int32         
 15  month_posted  51525 non-null  int32 

In [100]:
col = car_df.pop('make')
car_df.insert(2, 'make', col)
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   price         51525 non-null  int64         
 1   model_year    51525 non-null  int64         
 2   make          51525 non-null  object        
 3   model         51525 non-null  object        
 4   condition     51525 non-null  object        
 5   cylinders     51525 non-null  int64         
 6   fuel          51525 non-null  object        
 7   odometer      51525 non-null  int64         
 8   transmission  51525 non-null  object        
 9   type          51525 non-null  object        
 10  paint_color   51525 non-null  object        
 11  is_4wd        51525 non-null  int64         
 12  date_posted   51525 non-null  datetime64[ns]
 13  days_listed   51525 non-null  int64         
 14  year_posted   51525 non-null  int32         
 15  month_posted  51525 non-null  int32 

In [101]:
col = car_df.pop('days_listed')
car_df.insert(16, 'days_listed', col)
car_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   price         51525 non-null  int64         
 1   model_year    51525 non-null  int64         
 2   make          51525 non-null  object        
 3   model         51525 non-null  object        
 4   condition     51525 non-null  object        
 5   cylinders     51525 non-null  int64         
 6   fuel          51525 non-null  object        
 7   odometer      51525 non-null  int64         
 8   transmission  51525 non-null  object        
 9   type          51525 non-null  object        
 10  paint_color   51525 non-null  object        
 11  is_4wd        51525 non-null  int64         
 12  date_posted   51525 non-null  datetime64[ns]
 13  year_posted   51525 non-null  int32         
 14  month_posted  51525 non-null  int32         
 15  day_posted    51525 non-null  int32 

## EDA

In [102]:
car_df.sample(50)

price  model_year        make                        model  condition  \
51489   1200        2005  volkswagen             volkswagen jetta       fair   
11515   7700        2012       honda               honda civic lx  excellent   
38991  15375        2017      toyota                 toyota camry   like new   
30360  28995        2006         ram                     ram 3500       good   
27230   5995        2008   chevrolet             chevrolet impala       good   
19050   6000        2005   chevrolet     chevrolet silverado 1500       good   
41432  23998        2015        jeep          jeep grand cherokee  excellent   
20518  31000        2013         ram                     ram 2500  excellent   
32371  35000        1972   chevrolet           chevrolet corvette  excellent   
44193   3999        2006  volkswagen            volkswagen passat       good   
14775   6200        2016      nissan                nissan sentra  excellent   
2584   10995        2008        jeep                jeep wrangler  excellent   
32961  22400        2015        jeep                jeep wrangler  excellent   
33937  17995        2010   chevrolet   chevrolet silverado 2500hd  excellent   
45396   7500        2007         bmw                       bmw x5       good   
33586  10800        2013         ram                     ram 1500   like new   
17897   8000        2009        jeep          jeep grand cherokee  excellent   
12675   5707        2007        ford               ford fusion se  excellent   
15573   5900        2013     hyundai              hyundai elantra   like new   
3131    1650        2003        jeep   jeep grand cherokee laredo       fair   
34439   6495        2005      toyota                 toyota camry  excellent   
3592   16950        2018      nissan              nissan frontier   like new   
32940   5495        2009       honda                 honda accord       good   
5280    5000        2009      toyota                toyota tacoma       fair   
46407  10500        2004         gmc              gmc sierra 1500  excellent   
36651   3500        2005   chevrolet        chevrolet trailblazer   like new   
45813   6500           1      toyota                 toyota prius  excellent   
37347   3995        2007     hyundai             hyundai santa fe  excellent   
7339    8200        2014      toyota                 toyota camry  excellent   
9012    5700        2001        jeep                jeep wrangler  excellent   
37492   3988           1        ford                   ford f-150       good   
28078  25500        2013         ram                     ram 2500       good   
45632  13495        2016         kia                     kia soul   like new   
29608  26000        2017        ford                   ford f-150  excellent   
3366    7800        2008   chevrolet          chevrolet silverado       good   
49726  27500        2005         ram                     ram 3500  excellent   
46976  21000        2018      nissan  nissan frontier crew cab sv       good   
25692   1800        1998        ford                   ford f-150       fair   
22442  11900           1         gmc                   gmc acadia  excellent   
46247   2200        2003         ram                     ram 1500       good   
38884   1500        2013   chevrolet              chevrolet tahoe   like new   
9464       1        2018      nissan                 nissan rogue  excellent   
19574   3999        2007    chrysler      chrysler town & country  excellent   
24979   9900        2009         ram                     ram 1500   like new   
37661  19995        2006         gmc            gmc sierra 2500hd       good   
44107  10500        2007   chevrolet           chevrolet suburban   like new   
7847   26500        2014        ford                   ford f-250       good   
19116    287        2018   chevrolet             chevrolet impala   like new   
18786   4000        2000        ford         ford f250 super duty       good   
44948  

In [103]:
model_year_group = car_df.groupby('model_year')
make_group = car_df.groupby('make')
model_group = car_df.groupby('model')
condition_group = car_df.groupby('condition')

In [104]:
model_year_agg = model_year_group['price'].agg(['count','sum','max','min','mean','std']).reset_index()
model_year_agg

model_year  count       sum     max    min          mean           std
0            1   3619  42566009  109999      1  11761.815142   9465.731785
1         1908      2     25990   12995  12995  12995.000000      0.000000
2         1929      1     18000   18000  18000  18000.000000           NaN
3         1936      1      5000    5000   5000   5000.000000           NaN
4         1948      1     21000   21000  21000  21000.000000           NaN
..         ...    ...       ...     ...    ...           ...           ...
64        2015   3323  61143094  300000      1  18399.968101  10729.011701
65        2016   2954  57813498   76000      1  19571.258632  10408.650760
66        2017   2419  50163834   68000      1  20737.426209  10804.445977
67        2018   2193  53763964   90577      1  24516.171455  14970.534041
68        2019    380   9976589   84900      1  26254.181579  16482.045920

[69 rows x 7 columns]

In [105]:
count_vs_model_year_old_fig = px.line(model_year_agg, x='model_year', y='count', range_x=[1908,1990], range_y=[0,60], title='Number of Listed Cars by Model Year, Before 1991')
count_vs_model_year_old_fig.show()

In [106]:
count_vs_model_year_new_fig = px.line(model_year_agg, x='model_year', y='count', range_x=[1991,2019], title='Number of Listed Cars by Model Year, After 1990')
count_vs_model_year_new_fig.show()

In [107]:
price_vs_model_year_fig = px.line(model_year_agg, x='model_year', y='mean', range_x=[1908,2019], error_y='std', title='Mean Listed Price by Model Year')
price_vs_model_year_fig.show()

In [108]:
old_car_df = car_df.loc[car_df['model_year'] < 1960]
muscle_car_df = car_df.loc[(car_df['model_year'] >= 1960) & (car_df['model_year'] < 1980)]
pre_new_car_df = car_df[(car_df['model_year'] >= 1980) & (car_df['model_year'] < 2000)]
new_car_df = car_df[car_df['model_year'] >= 2000]

In [109]:
old_car_group = old_car_df.groupby(['paint_color', 'make'])['model'].count().reset_index()
old_car_group.columns = ['paint_color', 'make', 'count']
old_car_group.sort_values(['paint_color', 'make'])

paint_color       make  count
0         black      acura      6
1         black        bmw      3
2         black   cadillac      8
3         black  chevrolet    114
4         black   chrysler     10
..          ...        ...    ...
176      yellow       ford      6
177      yellow      honda      1
178      yellow    hyundai      1
179      yellow       jeep      4
180      yellow        ram      1

[181 rows x 3 columns]

In [110]:

old_car_hist = px.bar(old_car_group, x='paint_color', y='count', color='make', title='Number of Cars of Each Color, 1900-1959')
old_car_hist.show()

In [111]:
old_car_scat = px.scatter(old_car_df.sort_values(['paint_color', 'make']), x='paint_color', y='price', color='make', title='Listed Price in USD vs Color, 1900-1959')
old_car_scat.show()

In [112]:
muscle_car_group = muscle_car_df.groupby(['paint_color', 'make'])['model'].count().reset_index()
muscle_car_group.columns = ['paint_color', 'make', 'count']
muscle_car_group.sort_values(['paint_color', 'make'])
muscle_car_hist = px.bar(muscle_car_group, x='paint_color', y='count', color='make', title='Number of Cars of Each Color, 1960-1979')
muscle_car_hist.show()

In [113]:
muscle_car_scat = px.scatter(muscle_car_df.sort_values(['paint_color', 'make']), x='paint_color', y='price', color='make', title='Listed Price in USD vs Color, 1960-1979')
muscle_car_scat.show()

In [114]:
pre_new_car_group = pre_new_car_df.groupby(['paint_color', 'make'])['model'].count().reset_index()
pre_new_car_group.columns = ['paint_color', 'make', 'count']
pre_new_car_group.sort_values(['paint_color', 'make'])
pre_new_car_hist = px.bar(pre_new_car_group, x='paint_color', y='count', color='make', title='Number of Cars of Each Color, 1980-1999')
pre_new_car_hist.show()

In [115]:
pre_new_car_scat = px.scatter(pre_new_car_df.sort_values(['paint_color', 'make']), x='paint_color', y='price', color='make', title='Listed Price in USD vs Color, 1980-1999')
pre_new_car_scat.show()

In [116]:
new_car_group = new_car_df.groupby(['paint_color', 'make'])['model'].count().reset_index()
new_car_group.columns = ['paint_color', 'make', 'count']
new_car_group.sort_values(['paint_color', 'make'])
new_car_hist = px.bar(new_car_group, x='paint_color', y='count', color='make', title='Number of Cars of Each Color, 2000-2019')
new_car_hist.show()

In [117]:
new_car_scat = px.scatter(new_car_df.sort_values(['paint_color', 'make']), x='paint_color', y='price', color='make', title='Listed Price in USD vs Color, 2000-2019')
new_car_scat.show()